src/coreclr/vm/threads.cpp

```diff
@@ -2895,16 +2939,18 @@ void Thread::OnThreadTerminate(BOOL holdingLock)
     DWORD CurrentThreadID = pCurrentThread?pCurrentThread->GetThreadId():0;
     DWORD ThisThreadID = GetThreadId();
 
-#ifdef FEATURE_COMINTEROP_APARTMENT_SUPPORT
-    // If the currently running thread is the thread that died and it is an STA thread, then we
-    // need to release all the RCW's in the current context. However, we cannot do this if we
-    // are in the middle of process detach.
     if (!IsAtProcessExit() && this == GetThreadNULLOk())
     {
+#ifdef FEATURE_COMINTEROP_APARTMENT_SUPPORT
+        // If the currently running thread is the thread that died and it is an STA thread, then we
+        // need to release all the RCW's in the current context. However, we cannot do this if we
+        // are in the middle of process detach.
         CleanupCOMState();
-    }
 #endif // FEATURE_COMINTEROP_APARTMENT_SUPPORT
 
+        Thread::CleanUpForManagedThreadInNative(this);
```

Comment:
Is this still necessary to call this here? I do not think it will ever do anything useful. We take care of all situations of all situations that I can think of elsewhere.